src/types/basic/uint.ts

```diff
@@ -128,6 +134,96 @@ export class NumberUintType extends UintType<number> {
   }
 }
 
+const TWO_POWER_32 = 2 ** 32;
+
+/**
+ * For 64 bit number, we want to operator on HashObject
+ * over bytes to improve performance.
+ */
+export class Number64UintType extends NumberUintType {
+  constructor() {
+    super({byteLength: 8});
+    this._typeSymbols.add(NUMBER_64_UINT_TYPE);
+  }
+
+  /**
+   * TODO: move this logic all the way to persistent-merkle-tree?
+   * That's save us 1 time to traverse the tree in the applyDelta scenario
+   */
+  struct_deserializeFromHashObject(data: HashObject, numberOffset: number): number {
+    // a chunk contains 4 items
+    if (numberOffset < 0 || numberOffset > 3) {
+      throw new Error(`Invalid numberOffset ${numberOffset}`);
+    }
+    let low32Number = 0;
+    let high32Number = 0;
+
+    switch (numberOffset) {
+      case 0:
+        low32Number = data.h0 & 0xffffffff;
```

Comment:
Is it really necessary to mask with `0xffffffff`? Aren't we sure the number is already 32 bits?

Reply:
yeah it's not really necessary, it's just a pattern that make me feel safe and comfortable like everywhere else in ssz. It's a cheap operation anyway so I prefer keeping it.